x-pack/plugins/spaces/public/suspense_error_boundary/suspense_error_boundary.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiLoadingSpinner } from '@elastic/eui';
+import type { PropsWithChildren } from 'react';
+import React, { Component, Suspense } from 'react';
+
+import { i18n } from '@kbn/i18n';
+import type { NotificationsStart } from 'src/core/public';
+
+interface Props {
+  notifications: NotificationsStart;
+}
+
+interface State {
+  error: Error | null;
+}
+
+export class SuspenseErrorBoundary extends Component<PropsWithChildren<Props>, State> {
+  state: State = {
+    error: null,
+  };
+
+  static getDerivedStateFromError(error: Error) {
+    // Update state so next render shows fallback UI.
+    return { error };
+  }
+
+  public componentDidCatch(error: Error, { componentStack }: React.ErrorInfo) {
+    const { notifications } = this.props;
+    if (notifications) {
+      const title = i18n.translate('xpack.spaces.uiApi.errorBoundaryToastTitle', {
+        defaultMessage: 'Failed to load Kibana asset',
+      });
+      const toastMessage = i18n.translate('xpack.spaces.uiApi.errorBoundaryToastMessage', {
+        defaultMessage: 'Reload page to continue.',
+        values: { componentStack },
+      });
```

Comment:
In the absence of a fallback, can we add a reload button inside the toast so that users can click that instead of having to use browser UI?

Reply:
Good idea!

Reply:
Actually, we don't have the ability to add any elements / buttons:
https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/core/public/notifications/toasts/toasts_api.tsx#L58-L70

This should be a corner case error that user's don't see too often, so I think it's okay to leave it as-is, even if it's not ideal.